core/jobs/batch_jobs/exploration_population_jobs_test.py

```diff
@@ -198,3 +221,21 @@ def test_migrated_exploration_is_not_migrated(self) -> None:
             self.EXP_1_ID)
         self.assertEqual(
             unmigrated_exploration_model.android_proto_size_in_bytes, 64)
+        self.check_exploration_related_models()
+
+    def check_exploration_related_models(self) -> None:
+        """Check ExpSummaryModel and ExplorationRightsModel are
+        same as mentioned in the setup function.
+        """
+        exp_summary_model = exp_models.ExpSummaryModel.get_by_id(self.EXP_1_ID)
+        self.assertEqual(self.exp_summary_model, exp_summary_model)
+        self.assertEqual(self.exp_summary_model.version, 1)
+
+        exp_rights_snapshot_id = (
+            exp_models.ExplorationRightsModel.get_snapshot_id(self.EXP_1_ID, 1))
+        self.assertIsNotNone(
+            exp_models.ExplorationRightsSnapshotMetadataModel.get_by_id(
+                exp_rights_snapshot_id))
+        self.assertIsNotNone(
+            exp_models.ExplorationRightsSnapshotContentModel.get_by_id(
+                exp_rights_snapshot_id))
```

Comment:
One other thing to investigate perhaps: if you create an exploration in the develop branch, then publish it, then switch to the new code (without shutting down the server), then access the old version of the exploration, does that lead to any errors?

(You can access older versions by adding a "v" parameter in the exploration player URL -- see the reader.py code for details.)

Reply:
I didn't get the part `then access the old version`

The steps I just performed:
1. Create and publish exploration from the develop branch
2. Switch the branch to this pr
3. Fetch the exploration created in step 1
4. Edit the exploration and publish it again

Everything seems to work fine.

Reply:
To access the old version: make at least a couple commits first before switching to this PR (so that the exploration is on version 2 or 3 or so), then switch to this PR and go to the learner view (/explore/{{exp_id}}?v=1). The v=1 at the end references the old version.

Could you please try that and report the results? This is what I mean by testing previous versions.

Reply:
Tested it.

Steps I had performed:
1. Create an exploration from the develop branch example link - `http://localhost:8181/create/EfgKPBI39vGl#/gui/Introduction`
2. Made a couple of commits (version = 1 to 4)
3. Switch the code to this PR
4. Fetch each version `http://localhost:8181/explore/EfgKPBI39vGl#/?v=1`, `http://localhost:8181/explore/EfgKPBI39vGl#/?v=2` .. till 4 and all the commits are there in the same version order.